client/src/tests.rs

```diff
@@ -81,19 +80,28 @@ fn test_full_sync() {
         }
     }
 
-    let nodes = network(&[config("olive"), config("alice"), config("misha")]).unwrap();
+    let mut nodes = network(&[config("olive"), config("alice"), config("misha")]).unwrap();
     let (handle, _, _) = nodes.last().unwrap();
     let headers = BITCOIN_HEADERS.tail.clone();
     let height = headers.len() as Height;
     let hash = headers.last().unwrap().block_hash();
 
+    // Ensure all peers are connected to misha,
+    // so that misha can send effectively send blocks to
+    // all peers on time.
+    handle.wait_for_peers(2).unwrap();
+
     handle
         .import_headers(headers)
         .expect("command is successful")
         .expect("chain is valid");
 
-    for (node, _, thread) in nodes.into_iter() {
+    for (node, _, _) in nodes.iter_mut() {
+        node.set_timeout(std::time::Duration::from_secs(5));
         assert_eq!(node.wait_for_height(height).unwrap(), hash);
+    }
+
+    for (node, _, thread) in nodes.into_iter() {
```

Comment:
Splitting the loop is actually not necessary with the fix.

Reply:
Done